icu4c/source/common/umutex.h

```diff
@@ -20,38 +20,22 @@
 #ifndef UMUTEX_H
 #define UMUTEX_H
 
+#include <atomic>
+#include <condition_variable>
+#include <mutex>
+
 #include "unicode/utypes.h"
 #include "unicode/uclean.h"
-#include "putilimp.h"
-
-
-
-// Forward Declarations. UMutex is not in the ICU namespace (yet) because
-//                       there are some remaining references from plain C.
-struct UMutex;
-struct UConditionVar;
+#include "unicode/uobject.h"
 
-U_NAMESPACE_BEGIN
-struct UInitOnce;
-U_NAMESPACE_END
+#include "putilimp.h"
 
-/****************************************************************************
- *
- *   Low Level Atomic Operations.
- *      Compiler dependent. Not operating system dependent.
- *
- ****************************************************************************/
-#if defined(U_USER_ATOMICS_H)
-// Support for including an alternate implementation of atomic operations has been withdrawn.
+#if defined(U_USER_ATOMICS_H) || defined(U_USER_MUTEX_H)
```

Comment:
Why leave checks for U_USER_ATOMICS_H but not the other flags like U_NO_PLATFORM_ATOMICS, U_PLATFORM_IMPLEMENTS_POSIX, etc?

Reply:
Leaving as is. I don't want to clutter the code with more #ifdef checks.